content/local-setup.md

```diff
@@ -0,0 +1,283 @@
+
+First, open a terminal and check the version of Solana CLI by running the `solana --version` command.
+
+```bash
+solana --version
+```
+
+Next, to see the current Solana CLI configuration settings, run the `solana config get` command.
+
+```bash
+solana config get
+```
+
+From the list of outputs, the main ones to note are:
+
+- `RPC URL` - the endpoint you are using (connecting you to Localhost, Devnet, or Mainnet)
+- `Keypair Path` - the keypair used when running CLI commands (`id.json` is created by default)
+
+To change the `RPC URL`, use the `solana config set --url` command.
+
+```bash
+solana config set --url http://localhost:8899
+```
+
+```bash
+solana config set --url https://api.devnet.solana.com
+```
+
+```bash
+solana config set --url https://api.mainnet-beta.solana.com
+```
+
+You can generate a new keypair using the `solana-keygen new --outfile` command followed by the file path to store the keypair.
+
+```bash
+solana-keygen new --outfile ~/<FILE_PATH>
+```
+
+To update the `Keypair Path` for the Solana CLI config, using the `solana config set --keypair` command and specify the path of the new keypair (Solana CLI will now use the keypair from the specified path when running commands).
+
+```bash
+solana config set --keypair ~/<FILE_PATH>
+```
+
+To view the `publickey` of the currently configured keypair, run the `solana address` command.
+
+```bash
+solana address
+```
+
+To view the SOL balance of the of the currently configured keypair, run the `solana balance` command.
+
+```bash
+solana balance
+```
+
+To airdrop SOL for testing (on Devnet or Localhost), run the `solana airdrop <AMOUNT>` command (Note that while on Devnet you are limited to 2 SOL per airdrop).
+
+```bash
+solana airdrop 2
+```
+
+While deploying and testing on Localhost, you must run a test validator using the `solana-test-validator` command.
+
+```bash
+solana-test-validator
+```
+
+While on Localhost, you can view program logs by running the `solana logs` command in a separate terminal.
+
+```bash
+solana logs
+```
+
+To view program logs while on Devnet, use the `solana logs | grep "<PROGRAM_ID> invoke"` command and specify the program Id. If the program Id is not specified, then the `solana logs` command will return a constant stream of logs from Devnet. You can then specify the number of lines (of program logs) to return by including `-A <NUMBER_OF_LINES_TO_RETURN>` after the command.
+
+```bash
+solana logs | grep "<PROGRAM_ID> invoke" -A <NUMBER_OF_LINES_TO_RETURN>
+```
+
+To deploy a program, you using the `solana program deploy` command and specify the file path of the program. The command to deploy a program along with the file path is provided as an output when you build the program. We’ll go over this shortly in the demo.
+
+```bash
+solana program deploy <PATH>
+```
+
+Lastly, you can view the details for Solana CLI commands using the `solana --help` command.
+
+```bash
+solana --help
+```
+
+## Solana Program from Local Environment
+
```

Comment:
Need a little more lead-in here I think. "While the Solana Playground is enormously helpful, it's hard to beat the flexibility of your own local development environment. As you build more complex programs that are potentially integrated with one or more clients that are also under development, you may want to write and build your programs locally...."

Reply:
Same comment on this section as the last - it reads like the demo as opposed to an overview.